core/src/token_info/cached.rs

```diff
@@ -33,6 +36,25 @@ impl TokenInfoCache {
             cache: RwLock::new(cache),
         }
     }
+
+    /// Attempt to retrieve and cache all token info that is not already cached.
+    /// Fails if `all_ids` fails. Does not fail if individual token infos fail.
+    pub async fn cache_all(&self, number_of_parallel_requests: usize) -> Result<()> {
+        stream::iter(self.all_ids().await.context("failed to get all ids")?)
+            .for_each_concurrent(number_of_parallel_requests, |token_id| async move {
+                // Individual tokens might not conform to erc20 in which case we are unable to retrieve
+                // their info.
+                if let Err(err) = self.get_token_info(token_id).await {
+                    log::info!(
```

Comment:
Maybe `log::warn` so it gets a bit more attention? I imagine if we run into tokens that have this issue we will need to configure the token data with the correct decimals and such for it.